tiledb/sm/array/array.cc

```diff
@@ -146,8 +146,14 @@ Status Array::open_without_fragments(
     RETURN_NOT_OK(
         rest_client->get_array_schema_from_rest(array_uri_, &array_schema_));
   } else {
-    RETURN_NOT_OK(storage_manager_->array_open_for_reads_without_fragments(
-        array_uri_, *encryption_key_, &array_schema_));
+    auto&& [st, array_schema, array_schemas] =
+        storage_manager_->array_open_for_reads_without_fragments(
+            array_uri_, *encryption_key_);
+    if (!st.ok())
+      return st;
```

Comment:
For this specific one, and with the old way of just returning `Status`, we could have used `RETURN_NOT_OK(st)`. But I am assuming this will go away soon after we modify the return format of this function (and all functions) later on.